q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import yaml
+from glob import glob
+import pandas as pd
+from os.path import basename, isdir, splitext
+
+from q2_metadata.normalization._norm_messages import warningsCollection, errorsCollection
+
+
+class Rules(object):
+    """Collect the rules of one current variable rules.
+
+    Parameters
+    ----------
+    variable_rules_fp : str
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
+
+    remap : dict
+        Apply the rules of remapping variables values, by replacing
+        all instances of a given value by another one value.
+                    'remap:
+                      str: str'
+    validation : dict
+        Apply the rules of validating variables values, by replacing by the blank
+        (or the missing) value every instance which in one or many other columns
```

Comment:
im not sure i understand what's being said here

Reply:
see conversation above. Is that better? I may add a Glossary in the docstring to make the terms clearer: variable, value, sample, sample entry, etc. Do you like the idea?

Reply:
I will make a Glossary - as the end, unless told otherwise.

Reply:
I will make a glossary later - unless told otherwise